src/app/components/Button.js

```diff
@@ -14,9 +14,9 @@ export const Button = (
   }
   btnDomObj.innerText = btnObj.btnText;
   btnDomObj.classList.add('button');
-  btnDomObj.classList.add(...btnObj.classList);
+  if (btnObj.classList[0]) btnDomObj.classList.add(...btnObj.classList);
   btnDomObj.onclick = btnObj.onClickFn;
-
+  if (btnObj.isSpecial) btnDomObj.classList.add('button--special');
```

Comment:
O ile na jakiś tutorialach tak robią, to jednak nie jest to rekomendowana praktyka. Jeden if jeszcze ujdzie, ale to tzw. "rozdowjenie jazni w kodzie". Czyli Button sam nie wie czy jest special buttonem czy nie. Można tego ifa wyeliminować - im mniej ifow tym lepiej :)

Reply:
Czyli zamiast dodawać parametr i sprawdzać możesz zrobić tak. Proponuje: wywalić ten if z Button i zrobić nowy komponent np. 

const SpecialButton = (/*te same parametry co button*/ params) => {
    const button = Button(...params);
    button.classList.add('button--special');
   return button.
}

Reply:
Tak jak rozmawialiśmy - słuszna uwaga!
Rozważę to! Choć jeszcze w tym przypadku chyba będę skłaniał się nad opcją tą już dodaną, ale na przyszłość chyba tak zrobię!

Reply:
Tak ucza na tutorialach, ale tutoriale w większości nie są dobre :) Pokazują możliwość - ale nie najlepszą. 
Korzystałbym z moich doświadczeń w tym przypadku. Ale jak chcesz to już tutaj zostaw.

A tak z też ciekawości - zastanawia mnie - a Ty SCSSa pewnie znasz lepiej (a przede wszystkim lubisz!). Na poziomie SCSS też dałoby się to rozgraniczyć? Bo można np. jeszcze raz zadekralować chyba ten sam selektor co nie :) ? To mógłby być taki na special buttona tez osobny.

Reply:
> Popraw przynajmniej, żeby test przechodził :)

Testy naprawione!!!

Reply:
Jak najbardziej da radę, ale wydaje mnie się, że byłby to gwałt na BEM :D
Wtedy trochę nieBEMowo się robi i jest mniej czytelnie.